packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -151,6 +156,39 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
             ;
     }
 
+    renderYAxis(): React.ReactNode {
+        // Y axis with D3
+        const chartHeight = parseInt(this.props.style.height.toString());
+        const margin = { top: 15, right: 0, bottom: 5, left: this.props.style.yAxisWidth };
+        let allMax = 0;
+        this.state.xyData?.datasets?.forEach((dSet: any) => {
+            const rowMax = Math.max(...dSet.data);
+            allMax = Math.max(allMax, rowMax);
+        });
+        const yScale = scaleLinear()
+            .domain([0, Math.max(allMax, 1)])
+            .range([chartHeight - margin.bottom, margin.top]);
```

Comment:
It appears that when all values are the same e.g. allMax, then Chart.js will draw a line at the middle height. So the yScale has to be different.

Also, when all values are non-zero, then max is at top and min is at bottom, but the yScale is [0, max]. It should be [min, max].

Reply:
Works well now!